### PR TITLE
move & improve note about choosing between the two kinds of instructions

### PR DESCRIPTION
**Pull request by DrDaveD**

### CODE REVIEW

File: docs/worker-node/install-singularity.md [hunk 1]
```diff
@@ -36,14 +36,26 @@ available as at least a security update for all RHEL7-based releases.
...
 
-!!! note "Applicable versions"
-    The applicable software versions for this document are OSG version 3.4.2 or later.
-    The version of singularity installed should be 2.3.1 or later.
+!!! note
+    There are two separate sets of instructions on this page,
+    one for [installing privileged singularity](#installing-singularity-as-privileged)
+    and one for [enabling unprivileged singularity](#enabling-unprivileged-mode-for-singularity).
 
+    As of February 2018, no VO in the OSG is ready to invoke non-setuid Singularity in production without a worker node installation.
```
Comment (brianhlin) [inline docs/worker-node/install-singularity.md]: "Invoke" sounds overly dry here, let's just use "use"

File: docs/worker-node/install-singularity.md [hunk 2]
```diff
@@ -36,14 +36,26 @@ available as at least a security update for all RHEL7-based releases.
...
+    Most sites will want to follow the privileged RPM install instructions until there is wider VO support.
+    You may additionally want to enable unprivileged singularity for testing.
+
+    It is possible to both install the RPM and enable unprivileged singularity, then disable the privileges in the RPM, without the VO being involved.
+    You can do that by removing the '-suid' files in the RPM and setting the singularity.conf option to disable setuid.
+    You will be responsible for removing the '-suid' files again each time the RPM is installed and merging in any changes from the RPM's singularity.conf.
```
Comment (brianhlin) [inline docs/worker-node/install-singularity.md]: These instructions sound like a hack and I'm not entirely comfortable recommending this

**Comment by DrDaveD** [comment]
@brianhlin Please re-review

**Comment by brianhlin** [comment]
I've pushed a few commits. If we're going to mention unpriv singularity in RPM installs, we should provide instructions. Probably as a sub-section under "Optional configuration"

**Comment by DrDaveD** [comment]
I am OK with dropping the applicable versions note, but not with dropping the part of the note about following both sets of instructions.  That note is all about helping people to decide which set of instructions to follow, and I think that doing both is an important option.

If we add another set of instructions on modifying the rpm to work unprivileged, it will not make this note any less complicated because then it will also need to refer to that option.  I thought it would be better to just say it was not recommended.  However if you think we need that third set of instructions, let me know and I will add it and update the note.

**Comment by brianhlin** [comment]
I didn't see any note for following both instructions (both being RPM and CVMFS installations), just a note about unpriv singularity with an RPM installation. I don't see why we would need to mention it if we're not recommending it or providing instructions for it.

**Comment by DrDaveD** [comment]
The part about following both which you deleted was

> You may additionally want to enable unprivileged singularity for testing.

The reason for mentioning using the rpm unprivileged is because that is an option that people are likely to think of on their own and I want to warn about the downsides and say why we don't list the option.  Fermilab is already doing it and others may want to as well.

**Comment by brianhlin** [comment]
If people want to mess around with RPM-owned files that aren't `%config(noreplace)` and [possibly] shoot themselves in the foot, that's their prerogative. We don't provide that sort of disclaimer anywhere else in our docs and I don't see why we need to start here.

**Comment by DrDaveD** [comment]
In this case it isn't necessarily obvious, because there is a config option in the rpm that allows disabling setuid, and that does not require modifying the rpm.  However doing only that gives no security advantage because the setuid binaries are still present and so the risk is still there.  I think there should be a warning somewhere.  It could be at the end of the rpm section instead if that's better for you, but I thought it was appropriate to be up front where people are deciding which instructions to follow.

**Comment by bbockelm** [comment]
So, there's quite a bit of things we could warn people not to do -- the list is likely longer with HTCondor-CE, for example, than Singularity.  Let's look at existence proofs first rather than second-guess what people may potentially do wrong.  [FWIW - I disagree with the assertion that disabling setuid in the config file gives "no security advantage".  It provides _less advantage_ as it removes certain avenues of attack but leaves others.  That's an important distinction!]

Let's write the document to have two audiences:

1. The "normal sysadmin" who has been asked to install Singularity in support of their VO.  They should be able to read the introduction and clearly see they should install the privileged version.
2. The "beta tester" sysadmin who has self-selected as interested in enabling unprivileged mode as part of their VO's testing program.  Assume they need to learn how to make the kernel changes, then follow directions from their VO.  We still don't know how VOs will deploy this.

I do _not_ want the "normal sysadmin" to read this page and try to follow both as they end up increasing their security risks -- they're now vulnerable to both Singularity CVEs and kernel CVEs.

Once we're to the point where VOs use unprivileged mode in production, we can update the recommendations.

**Comment by DrDaveD** [comment]
@bbockelm OK, what does that mean practically?  How does it affect what we should put in this PR?  I think it already does a pretty good job of addressing the normal sysadmin, but probably not the beta tester.  Maybe you could push your own updates to the PR.

**Comment by DrDaveD** [comment]
I don't want to let this languish for a long time, leaving the instruction page as-is with no guidance about the two different types of install instructions.  How about we merge this PR and later have a separate one for the differences that @bbockelm would like to see.

**Pull request merged**